common/writeThoughFile.go

```diff
@@ -71,7 +75,16 @@ func CreateDirectoryIfNotExist(directory string, tracker FolderCreationTracker)
 			// consistent.
 			tracker.RecordCreation(directory)
 		}
-		return err
+
+		// another routine might have created the directory at the same time
+		// check whether the directory now exists
+		if _, err := OSStat(directory); err != nil {
```

Comment:
Do you wanna do this OSStat only if mkDirErr != nil?  Three reasons for that:
* when mkDirErr == nil, the test doesn't accomplish anything useful  (since we would have returned nil without the test anyway)
* hypothetically (I haven't tested it myself, but I think someone else has) Stat may fail but mkdir may succeed. (Due to being in a write-only situation.  So replacing nil mkdir error with a non-nill Stat err would be misleading in that case)
* It will save a disk IO operation

Reply:
Good call, I forgot a return statement